### PR TITLE
chore: Switch to the new partial execution logic in test runner (no-changelog)

### PR DESCRIPTION
**Pull request by burivuhster**
## Summary

To correctly start the execution of a workflow under test from the same trigger node as in a specific past execution, we need to switch to the new partial execution logic.
This PR changes the `partialExecutionVersion` for test executions.
It also uses the `Workflow` instance to figure out start nodes (children of a trigger node).

Also, PR contains minor refactoring of the tests (`mockNodeTypesData` was extracted to a separate file to be re-used in multiple tests).

## Related Linear tickets, Github issues, and Community forum posts

- https://linear.app/n8n/issue/AI-514/switch-to-the-new-partial-execution-logic-in-test-runner
- https://linear.app/n8n/issue/AI-493/figure-out-trigger-node-for-execution-and-start-evaluation-execution
- Previous work on the same topic was done here: #11983

## Review / Merge checklist

- [ ] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/cli/src/evaluation/test-runner/test-runner.service.ee.ts [hunk 1]
```diff
@@ -43,8 +45,45 @@ export class TestRunnerService {
...
+			connections: workflow.connections,
+			active: false,
+			nodeTypes: this.nodeTypes,
+		});
+
+		// Determine the trigger node of the past execution
+		const pastExecutionTriggerNode = getPastExecutionTriggerNode(pastExecutionData);
+		assert(pastExecutionTriggerNode, 'Could not find the trigger node of the past execution');
+
+		// Prepare the data structures
```
Comment (tomi) [inline packages/cli/src/evaluation/test-runner/test-runner.service.ee.ts:67]: I don't think this comment provides any value. We do see from the code that it intializes an object

```suggestion
```

File: packages/cli/src/evaluation/test-runner/test-runner.service.ee.ts [hunk 2]
```diff
@@ -43,8 +45,45 @@ export class TestRunnerService {
...
+		const triggerToStartFrom = {
+			name: pastExecutionTriggerNode,
+			data: pastExecutionData.resultData.runData[pastExecutionTriggerNode][0],
```
Comment (tomi) [inline packages/cli/src/evaluation/test-runner/test-runner.service.ee.ts:70]: Should we assert that this is indeed defined? Or does it need to be defined?

File: packages/cli/src/evaluation/test-runner/test-runner.service.ee.ts [hunk 3]
```diff
@@ -43,8 +45,45 @@ export class TestRunnerService {
...
+		};
+
+		// Start nodes are the nodes that are connected to the trigger node
+		const startNodes = workflowInstance
+			.getChildNodes(pastExecutionTriggerNode, NodeConnectionType.Main, 1)
+			.map((name) => ({
+				name,
```
Comment (tomi) [inline packages/cli/src/evaluation/test-runner/test-runner.service.ee.ts:77]: To be explicit what name this is:

```suggestion
			.map((nodeName) => ({
				name: nodeName,
```

File: packages/cli/src/evaluation/test-runner/test-runner.service.ee.ts [hunk 4]
```diff
@@ -43,8 +45,45 @@ export class TestRunnerService {
 		private readonly activeExecutions: ActiveExecutions,
 		private readonly testRunRepository: TestRunRepository,
 		private readonly testMetricRepository: TestMetricRepository,
+		private readonly nodeTypes: NodeTypes,
 	) {}
 
+	private getStartNodesData(
```
Comment (tomi) [inline packages/cli/src/evaluation/test-runner/test-runner.service.ee.ts:51]: Could we be more explicit about what data this returns, either by renaming the fn or adding a comment? Is it input data of the start nodes?

File: packages/cli/src/evaluation/test-runner/test-runner.service.ee.ts [hunk 5]
```diff
@@ -57,20 +96,13 @@ export class TestRunnerService {
...
-		// Determine the start node of the past execution
-		const pastExecutionStartNode = getPastExecutionTriggerNode(pastExecutionData);
-
 		// Prepare the data to run the workflow
 		const data: IWorkflowExecutionDataProcess = {
-			destinationNode: pastExecutionData.startData?.destinationNode,
-			startNodes: pastExecutionStartNode
-				? [{ name: pastExecutionStartNode, sourceData: null }]
-				: undefined,
+			...this.getStartNodesData(workflow, pastExecutionData),
```
Comment (tomi) [inline packages/cli/src/evaluation/test-runner/test-runner.service.ee.ts:101]: We now omit `destioationNode` entirely. Is that intentional?

File: packages/cli/test/integration/shared/utils/node-types-data.ts [hunk 6]
```diff
@@ -0,0 +1,33 @@
+import type { INodeTypeData } from 'n8n-workflow';
+
+export function mockNodeTypesData(
```
Comment (tomi) [inline packages/cli/test/integration/shared/utils/node-types-data.ts:3]: 👍 for moving to shared

**Review by tomi — COMMENTED** [review]
Couple questions

File: packages/cli/src/evaluation/test-runner/test-runner.service.ee.ts (hunk 5 above)
Comment (burivuhster) [reply packages/cli/src/evaluation/test-runner/test-runner.service.ee.ts:101]: Yes, we decided to run the entire wf, even if the past execution was partial (had destination node).

File: packages/cli/src/evaluation/test-runner/test-runner.service.ee.ts (hunk 4 above)
Comment (burivuhster) [reply packages/cli/src/evaluation/test-runner/test-runner.service.ee.ts:54]: I tried to add some context in the comment. This entire method is preparing a part of the large data structure that would be supplied to the workflow runner, to ensure the test execution will "re-play" the past execution from the same trigger node.

**Review by tomi — APPROVED** [review]
🚀

**Comment by janober** [comment]
Got released with [`n8n@1.73.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.73.0)

**Pull request merged**